bakery/codec.go

```diff
@@ -32,22 +26,27 @@ type caveatId struct {
 }
 
 // encodeJSONCaveatId creates a JSON encoded third-party caveat.
-func encodeJSONCaveatId(key *KeyPair, ci caveatInfo) ([]byte, error) {
+func encodeJSONCaveatId(
```

Comment:
Knowing the protocol, this makes sense, however, It would be nice to have an explanation why thirdPartyPub is needed.